FWCore/Framework/interface/ESProducer.h

```diff
@@ -229,7 +227,8 @@ namespace edm {
     ESConsumesCollectorT<TRecord> setWhatProduced(TFunc&& func, TDecorator&& iDec, const es::Label& iLabel = {}) {
       const auto id = consumesInfos_.size();
       using DecoratorType = std::decay_t<TDecorator>;
-      using CallbackType = eventsetup::Callback<ESProducer, TFunc, TReturn, TRecord, DecoratorType>;
+      using CallbackType = eventsetup::
+          Callback<ESProducer, eventsetup::DummyAcquireFunctor<TRecord>, TFunc, TReturn, TRecord, DecoratorType>;
```

Comment:
Could the common parts of this function and the `setWhatProducedExternalWork()` that also calls `registerProducts()` be abstracted in to a common function? IIUC the specific part is the actual `CallbackType` and how it is constructed, and the rest (`registerProducts()` and dealing with `consumesInfos_`) could be common?